include/fmt/color.h

```diff
@@ -209,18 +209,103 @@ struct rgb {
   uint8_t b;
 };
 
-void vprint_rgb(rgb fd, string_view format, format_args args);
-void vprint_rgb(rgb fd, rgb bg, string_view format, format_args args);
+namespace internal {
+FMT_CONSTEXPR_DECL const char FOREGROUND_COLOR[] = "\x1b[38;2;000;000;000m";
+FMT_CONSTEXPR_DECL const char BACKGROUND_COLOR[] = "\x1b[48;2;000;000;000m";
+
+template <typename Char>
+struct ansi_color_escape {
+  FMT_CONSTEXPR ansi_color_escape(rgb color, const char esc[]) FMT_NOEXCEPT {
+    for (int i = 0; i < 7; i++) {
+      buffer[i] = static_cast<Char>(esc[i]);
+    }
+    to_esc(color.r, buffer, 7, ';');
+    to_esc(color.g, buffer, 11, ';');
+    to_esc(color.b, buffer, 15, 'm');
+    buffer[19] = static_cast<Char>(0);
+  }
+  FMT_CONSTEXPR operator const Char *() const FMT_NOEXCEPT { return buffer; }
+
+private:
+  Char buffer[sizeof(FOREGROUND_COLOR)];
+
+  FMT_CONSTEXPR void to_esc(uint8_t c, Char out[], int offset,
```

Comment:
I suggest getting rid of the `offset` parameter and passing buffer pointer adjusted by offset instead.